### PR TITLE
[bazel] Bazelize the SCA targets

### PR DESCRIPTION
**Pull request by mcy**
Fixes: https://github.com/lowRISC/opentitan/issues/11509

Alphan, would you mind checking this is a sensible setup?

### CODE REVIEW

File: sw/device/lib/base/macros.h [hunk 1]
```diff
@@ -257,4 +257,13 @@
   do {                                                                       \
   } while (false)
 
+/**
+ * Evaluates and discards `expr_`.
+ *
+ * This is needed because ‘(void)expr;` does not work for gcc.
+ */
+#define OT_DISCARD(expr_) \
+  if (expr_) {               \
```
Comment (alphan) [inline sw/device/lib/base/macros.h:266]: nit: is this a clang-format issue?

**Review by alphan — APPROVED** [review]
Thanks @mcy! Couldn't check the CI results because it looks like the quick-lint failed.

File: sw/device/lib/base/macros.h (hunk 1 above)
Comment (mcy) [reply sw/device/lib/base/macros.h]: It is indeed. :)

**Comment by a-will** [comment]
@mcy I took a look. It seems this breaks the patch at [hw/top_englishbreakfast/util/sw_sources.patch](https://github.com/mcy/opentitan-1/blob/bazel-sca/hw/top_englishbreakfast/util/sw_sources.patch), leading to test failures on top_englishbreakfast. You'll probably need to regenerate the diff for `sw/device/sca/lib/sca.c`.

File: sw/device/sca/lib/BUILD [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        "//sw/device/lib/dif:gpio",
+        "//sw/device/lib/dif:rv_timer",
+        "//sw/device/lib/dif:uart",
+        "//sw/device/lib:irq",
+        "//sw/device/lib:pinmux",
+        "//sw/device/lib/runtime:hart",
+        "//sw/device/lib/runtime:print",
+        "//sw/device/lib/testing/test_framework:ottf_start",
+    ],
+)
```
Comment (vogelpi) [inline sw/device/sca/lib/BUILD:60]: missing newline

File: sw/device/sca/BUILD [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        deps = [
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/sca/lib:prng",
+        "//sw/device/lib/dif:kmac",
+        "//sw/device/lib/base:memory",
+        "//sw/device/lib/runtime:log",
+        "//sw/device/sca/lib:sca",
+        "//sw/device/sca/lib:simple_serial",
+    ],
+)
```
Comment (vogelpi) [inline sw/device/sca/BUILD:35]: missing newline

**Review by vogelpi — COMMENTED** [review]
Thanks for doing this @mcy. I just have some nits.

Maybe also @vrozic should have a quick look as he is currently working on extending the experimental setup for KMAC using this code.

**Pull request merged**